shared/actions/git.js

```diff
@@ -234,45 +130,72 @@ function _handleIncomingGregor(action: GitGen.HandleIncomingGregorPayload) {
   }
 }
 
-function _navigateToTeamRepo(action: GitGen.NavigateToTeamRepoPayload) {
-  return Saga.sequentially([
-    Saga.call(_loadGit, GitGen.createLoadGit()),
-    Saga.identity(action),
-    // This needs to be a select so we get the store post-loadGit
-    Saga.select((state: TypedState) => state.entities.getIn(['git', 'idToInfo'])),
-  ])
+const navToGit = (_, action: GitGen.NavToGitPayload) => {
+  const path = isMobile ? [Tabs.settingsTab, SettingsConstants.gitTab] : [Tabs.gitTab]
+  const actions = [Saga.put(navigateTo(path, []))]
+  if (action.payload.routeState) {
+    actions.push(Saga.put(setRouteState(path, action.payload.routeState)))
+  }
+  return Saga.all(actions)
 }
 
-function _processNavigateToTeamRepo(results: any[]) {
-  const {teamname, repoID} = (results[1]: GitGen.NavigateToTeamRepoPayload).payload
-  const idToInfo: I.Map<string, Types.GitInfo> = results[2]
+const isRepoInfoFresh = (lastLoad: ?number, maxAgeMs: number) => lastLoad && lastLoad - Date.now() < maxAgeMs
+
+// Note: Needs to be sequential (& possibly refire) in order to match the repoID + teamname to a unique id
+const navigateToTeamRepo = (state: TypedState, action: GitGen.NavigateToTeamRepoPayload) => {
```

Comment:
i think if you really need some kind of async logic it might as well be a small generator, we typically do this with a small call. i'd suggest

```
const navigateToTeamRepo = (state: TypedState, action: GitGen.NavigateToTeamRepoPayload) => Saga.call(function* (){
  const {teamname, repoID} = action.payload
  const idToInfo = state.git.idToInfo
  let repo = idToInfo.find(val => val.repoID === repoID && val.teamname === teamname)
  if (!repo) {
    // try and load once
      yield Saga.put(GitGen.createLoadGit())
      yield Saga.take(GitGen.loaded)
      const nextState = yield Saga.select()
      const idToInfo = nextState.git.idToInfo
       repo = idToInfo.find(val => val.repoID === repoID && val.teamname === teamname)
  }
... the rest
})

```
^ could be cleaned up by making some helpers so there's less copy pasta

also i don't think we need this 'stale' concept. this stuff doesn't change that often. just load if its not in there. you likely clicked cause someone did something recently so 1 hour might actually not be fresh enough.

this might be simpler than redoing the action.